internal/gamepaddb/gen.go

```diff
@@ -0,0 +1,183 @@
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build ignore
+
+// This program generates platform-specific controller dbs. It can be invoked by running
+// go generate
+package main
+
+import (
+	"bytes"
+	_ "embed"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+	"text/template"
+)
+
+const license = `// Copyright 2024 The Ebitengine Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+`
+
+// gamecontrollerdb.txt is downloaded at https://github.com/mdqinc/SDL_GameControllerDB.
+
+// To update the database file, run:
+//
+//     curl --location --remote-name https://raw.githubusercontent.com/mdqinc/SDL_GameControllerDB/master/gamecontrollerdb.txt
+
+//go:embed gamecontrollerdb.txt
+var gamecontrollerdb_txt []byte
+
+const db_template string = `
+{{.License}}
+
+{{.DoNotEdit}}
+
+{{.BuildTag}}
+
+package gamepaddb
+
+import _ "embed"
+
+//go:embed gamecontrollerdb_{{.Platform}}.txt
+var controllerBytes []byte
+{{if eq .Platform "windows" }} 
+//go:embed gamecontrollerdb_glfw.txt
+var glfwControllerBytes []byte
+{{end}}
+func init() {
+	if err := Update(controllerBytes); err != nil {
+		panic(err)
+	}
+{{if eq .Platform "windows" }} 
+    if err := Update(glfwControllerBytes); err != nil {
+        panic(err)
+    }
+{{end}}}`
+
+func main() {
+	// Follow the standard comment rule (https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source).
+	doNotEdit := "// Code generated by gen.go using 'go generate'. DO NOT EDIT."
+
+	type gamePadPlatform struct {
+		name     string
+		buildTag string
+	}
+
+	supported := map[string]gamePadPlatform{
+		"Windows": {
+			name:     "windows",
+			buildTag: "//go:build windows && !microsoftgdk",
+		},
+		"Mac OS X": {
+			name:     "macos_darwin",
+			buildTag: "//go:build darwin && !ios",
+		},
+		"Linux": {
+			name:     "linbsd",
+			buildTag: "//go:build (freebsd || (linux && !android) || netbsd || openbsd) && !nintendosdk && !playstation5",
+		},
+		"iOS": {
+			name:     "ios",
+			buildTag: "//go:build ios",
+		},
+		"Android": {
+			name:     "android",
+			buildTag: "//go:build android",
+		},
+	}
+
+	// divide controller dbs into chunks by platform
+	controllerDbs := splitControllerDb(gamecontrollerdb_txt)
```

Comment:
Use `DBs` instead of `Dbs` to follow the Go convention.

https://google.github.io/styleguide/go/decisions.html#initialisms